server.js

```diff
@@ -1,53 +1,193 @@
 // server.js
-const express = require('express');
-const serveStatic = require('serve-static');
-const history = require('./customHistory');
+const express = require("express");
+const serveStatic = require("serve-static");
+const history = require("./customHistory");
+const winston = require("winston");
+const { combine, timestamp, json, errors } = winston.format;
+const morgan = require("morgan");
+const rateLimit = require("express-rate-limit");
 
 const app = express();
 
+// =======================
+// Logger Configuration
+// =======================
+
+const logger = winston.createLogger({
+    level: process.env.LOG_LEVEL || "info",
+    format: combine(errors({ stack: true }), timestamp(), json()),
+    transports: [new winston.transports.Console({ handleExceptions: true })],
+    exitOnError: false,
+});
+
+logger.stream = { write: (message) => logger.info(message.trim()) };
+
+// =======================
+// Body Parser & Request Logging
+// =======================
+
+app.use(express.json({ limit: "10kb" }));
+app.use(morgan("combined", { stream: logger.stream }));
+
+// =======================
+// Rate Limiting
+// =======================
+
+// Can't exceed 100 logger requests within 1 minute
+const logLimiter = rateLimit({
+    windowMs: 1 * 60 * 1000,
+    max: 100,
+    message: {
+        status: 429,
+        error: "Too many logging requests from this IP, please try again after 1 minutes.",
+    },
+    standardHeaders: true,
+    legacyHeaders: false,
+});
+
+// =======================
+// Log Endpoint
+// =======================
+
+const redactApiKey = (apiKey) =>
+    apiKey ? apiKey.replace(/.(?=.{4})/g, "*") : "N/A";
```

Comment:
Nit: is this normal you're not using your function https://github.com/algolia/tools/pull/51/files?diff=split&w=1#diff-37e67fab62d89388360ca56bb6ab41bc8d1e852feab41978303e548652deee84R29 ?